clang-tools-extra/test/clang-tidy/checkers/modernize/use-designated-initializers.cpp

```diff
@@ -0,0 +1,130 @@
+// RUN: %check_clang_tidy -std=c++17 %s modernize-use-designated-initializers %t
```

Comment:
add test with agregate in agregate, like `x = {5, {1}};` and one with some implicit conversion like `{5, 1}`.